### DISCUSSION TITLE
LuaJIT integration

### DISCUSSION THREAD

**Discussion by HaronK** [mlua-rs/mlua#282] [Q&A]
Hi all,
I'm wondering how mlua is working with LuaJIT. Does it support LuaJIT feature of inlining a calls to C functions or is it just bounding via the classic  Lua/C API?

**Comment by khvzak** [comment]
> Does it support LuaJIT feature of inlining a calls to C functions

You mean through the `ffi` module integrated into luajit? Unfortuantely no, C calls are made through Lua C API and not JIT optimized.

I don't have any plans yet to make integration with luajit ffi. Personally I think luajit is sunsetting. Supporting it is a bit pain. It uses non-standard and complex build system. Cross compilation is non-trivial too. No clear versioning, etc, etc.

At the moment I look forward to JIT backend for Luau from Roblox. It still under development, but I added support in the master branch. If they introduce jit-optimized C functions very likely they will be supported in mlua too.
